Added basic help text functionality

utkarshcmu opened the pull request:
For screenshots, please refer #4079 

May be this PR needs some polishing, but otherwise it serves the basic functionality.

torkelo commented:
not sure about this implementation, maybe a more programmatic approach that used  thether drop more directly.  Need to keep use of directives and other bindings in panels titles to a low for performance reasons, https://github.com/grafana/grafana/blob/master/public/app/core/components/info_popover.ts#L35

utkarshcmu commented:
@torkelo Thanks for reviewing the PR. Will make the changes asap and ask for a review again. :)

joelanford committed:
Added support for TLS client auth for datasource proxies (#5801)

Commit:
Make tooltip shareable like crosshair

IllyaChekrygin committed:
Add VictorOps alert notification capability

IllyaChekrygin committed:
Fix user-facing text

IllyaChekrygin committed:
Change VictorOps fa from `fa-exclamation-triangle` -> `fa-pagelines`

IllyaChekrygin committed:
Add option to alert on ExecutionError

IllyaChekrygin committed:
Fix return on `nil` and cleanup debug messages

utkarshcmu committed:
Removed info-popover

utkarshcmu committed:
Removed info-popover

bergquist committed:
fix(influxdb): fixes broken raw query usage

bergquist committed:
tech(influxdb): refactor query builder

trying to reduce the amounts of moving parts for influxdb

bergquist committed:
feat(alertlist): make it possible to filter on alerts from current dashboard

bergquist committed:
tech(conf): remove dragoo :(

bergquist committed:
Merge pull request #6538 from grafana/alertlist_only_from_dashboard

feat(alertlist): make it possible to filter on alerts from current da…

bergquist committed:
fix(influxdb): return internal influxdb errors

ref #6523

Commit:
Fix for cloudwatch datasource requesting too many datapoints (#6544)

The range checking in _getPeriod appeared to be using a different variable than the period that had just been calculated for its bounds checks.

Commit:
Fix typo (password strenght -> password strength)

bergquist committed:
Merge pull request #6546 from huydx/master

Fix typo (password strenght -> password strength)

bergquist committed:
feat(stats_usage): add stats about alerts

bergquist committed:
feat(stats): add alerts to global admin stats

utkarshcmu committed:
Added OR to alert_tab

daniellee committed:
refactor(securejsondata): extract to class

Extract from pluginsettings class so that the
securejsondata type can be used in the other
classes. Encrypt and decrypt functions
extracted too.

bergquist committed:
fix(influxdb): add support for regex measurments

closes #6560

torkelo committed:
docs(): fixed ldap link

nicr9 committed:
docs(datasources/influxdb): Fix broken image link

Twas missing a leading `/`.

bergquist committed:
Merge pull request #6569 from nicr9/patch-1

docs(datasources/influxdb): Fix broken image link

ryantxu committed:
round the seconds

ryantxu committed:
missing space

torkelo committed:
docs(): updated changelog

bergquist committed:
feat(alerting): improve error logging for extracting alerts

ref #6576

utkarshcmu committed:
Implemented operator based firiing in backend

utkarshcmu committed:
Added tests for checking nested operators

Commit:
Correct series highlight (#6578)

solves issue #6573

tomkozlowski committed:
added explicitly setting token as Bearer Type

utkarshcmu committed:
Added OR condition in docs

bergquist committed:
docs(api): add docs about creating new org

close #6588

torkelo committed:
fix(panel): set initial transparency state at first link

daniellee committed:
refactor(dataproxy): TLS Client Auth

Use a SecureJsonData field for TLS
Client Auth instead of 3 new db
fields. Same model as used for
PluginSettings.

Saves and encrypts the pem file
content rather than just saving
the paths to the cert and key.
This allows for uploading from
the Edit Datasource page in
Grafana.

mattttt committed:
Removed green border from OK alerts. Makes dashboards too busy, competes for attention where action is unnceessary (#6555)

mtanda committed:
(cloudwatch) long retention support (#6547)

bergquist committed:
docs(notifications): fixes typo

daniellee committed:
feat(dataproxy): TLS CA Cert for self-signed certs

For self-signed TLS Certificates, authentication
with InsecureSkipVerify set to false then this
error will occur:

x509: certificate signed by unknown authority

The solution is to allow the user to upload the
CA cert as well.

torkelo committed:
fix(ux): fixed dropzone display issues, fixes #6598

torkelo committed:
fix(build): fixed failing unit test

torkelo committed:
ux(view/tv mode): treat tab change as user activity, resets in activity timer

torkelo committed:
fix(units): fixed issue with data rate bit units, fixes #6602

DanCech committed:
Merge pull request #6584 from tomkozlowski/bugfix/oauth-login-bearer-token

explicitly set oauth2 token to Bearer Type

DanCech committed:
fix up sass lint warnings (#6603)

torkelo committed:
fix(404): fixed 404 page

torkelo committed:
fix(error handling): fixed server side error handling page

torkelo committed:
cleanup(): removed unused uio-angular tab component

torkelo committed:
ux(tweak): minor polish to date picker styling

torkelo committed:
ux(tweak): minor polish to date picker styling

torkelo committed:
change(alerting): delete alert rule from panel model when duplicating panel

torkelo committed:
cleanup(): removed unused gfbox style

torkelo committed:
cleanup(sass): removed tight-form styles

torkelo committed:
fix(singletat): Support repeated template variables in prefix/postfix, fixes #6595

torkelo committed:
docs(): minor update to centos docs about font packages needed for phantomjs

torkelo committed:
testdata(): added test graph for null stacking scenario, #6566

IllyaChekrygin committed:
Resolve rebase w/ upstream conflicts.
Remove: `Alert on Exec Error` (no longer needed)
Add: `ruleUrl` to VictorOps message body.

IllyaChekrygin committed:
Add VictorOps Test.

IllyaChekrygin committed:
Fix formatting

torkelo committed:
fix(dashboard): minor fixes, and restored in-activity timer to 1min

utkarshcmu committed:
Added firingEvalution to Rule test

utkarshcmu committed:
Added braces to single condition firingEvaluation string

torkelo committed:
fix(dashboard): fixed issue when dragging new panel to drop zone

utkarshcmu committed:
Added tests for firingEvaluation string

torkelo committed:
fix(templating): fixed issue when adding template variable, fixes #6622

torkelo committed:
feat(templating): don't persist template variable options when variable has refresh enabled, closes #6586

torkelo committed:
fix(templating): work on fixing exporting issues when using templating variables, like data source variables, and panel repeats, requires more work #6189

torkelo committed:
Merge branch 'or_alerting' of https://github.com/utkarshcmu/grafana into utkarshcmu-or_alerting

bergquist committed:
fix(alerting): remove possible divide by zero panic

ref #6599

bergquist committed:
fix(influxdb): fixes broken tag rendering for influxdb alerting

closes #6626
ref #6523

torkelo committed:
refactor(alerting): refactoring PR for OR conditions,  #6579

torkelo committed:
Merge branch 'utkarshcmu-or_alerting'

torkelo committed:
changelog(): added entry for PR #6579

torkelo committed:
Merge branch 'master' of github.com:grafana/grafana

ryantxu committed:
Use ms rather than second precision

bergquist committed:
test(alerting): fixes broken unit test

Commit:
Revert #6287 for jquery.flot.stack.js

bergquist committed:
fix(settings): remove nil pointer exception

daniellee committed:
feat(datasources): allow updating of tls certs

The tls files are stored as a json blob in the
SecureJsonData field. To update one file, the
json has to be fetched from the db first and
then updated with the cert file that has been
changed.

Also, a change to the dto with flags that are
used in the frontend to show if a file has
been uploaded. For example, if tlsClientKeySet
is set to true then means a file for the
client key is saved in the db. This is to
avoid sending any private data to the frontend.

daniellee committed:
feat(httpsettings): add tls auth option

- Three text areas where the user can paste
in the CA Cert (optional), Client Cert
and Client Key.

- Tooltips for Auth checkboxes with brief
descriptions of what With Credentials and
With CA Cert are.

- Adds popover for TLS Auth header too.

- Aligns gf-form elements as labels and
checkboxes were not aligned before.

- Makes CA Cert optional as it is only
needed for self-signed certs.

Commit:
Revert #6287 for graph_tooltip.js

daniellee committed:
fix(dataproxy): test with CA Cert

Commit:
Correctly stack series with null and missing values

Commit:
Highlight correct point when series has null values

Commit:
Typo

Commit:
Come back to #6287 for graph_tooltip.js

as it is an only one loop solution

torkelo committed:
Merge branch 'issue-6566' of https://github.com/benrubson/grafana into benrubson-issue-6566

torkelo committed:
Merge branch 'master' of github.com:grafana/grafana

bergquist committed:
fix(notifications): failed image upload should not stop notification

closes #6638

torkelo committed:
bumped version to v4-beta2

torkelo committed:
docs(): updated doc download links

torkelo committed:
docs(): fixed windows link

torkelo committed:
Update CHANGELOG.md

bergquist committed:
docs(readme): add link to whats new in grafana 4

bergquist committed:
docs(issue_template): add note about alerting

bergquist committed:
fix(alerting): fixes bug that prevents noData from triggering

closes #6631

torkelo committed:
fix(dashboard): minor fix to schema upgrade when schemaVersion is not correct, fixes #6650

mtanda committed:
(cloudwatch) update aws-sdk-go to v1.5.8 (#6658)

* (cloudwatch) upgrade aws-sdk-go v1.5.8

* (cloudwatch) remove non-required aws-sdk-go part

* add go-ini

bergquist committed:
feat(alerting): add backend support for median reducer

torkelo committed:
fix(templating): fixed duplicate button bug, fixes #6652

bergquist committed:
feat(alerting): add median reducer to frontend

tomkozlowski committed:
generalized database connection cert support and added to postgres

peregrinogris committed:
Fix typo in Basic Concepts

bergquist committed:
Merge pull request #6671 from peregrinogris/patch-2

Fix typo in Basic Concepts

bergquist committed:
docs(influxdb): typo

dirkleroux committed:
first, delta, range passing unit tests

Signed-off-by: Dirk le Roux <dirk.leroux@gmail.com>

dpavlos committed:
[6486] Fix status code when adding an existing user to org (#6678)

torkelo committed:
fix(alerting): check if dashboard is saved before testing alert rule, fixes #6549

torkelo committed:
fix(server side rendering): Fixed address used when rendering panel via phantomjs and using non default http_addr config, fixes #6660

bergquist committed:
feat(cloudwatch): adds access and secret key to edit config page

torkelo committed:
fix(graph): fixed order of graph tooltip when sort order was none, fixes #6648

torkelo committed:
fix(streaming): clear data stream on subject completion

bergquist committed:
feat(cloudwatch): add authtype dropdown to config page

bergquist committed:
test(cloudwatch): fixes failing tests

dirk-leroux committed:
oAdded value type documentation to singlestat panel

Signed-off-by: Dirk le Roux <dirk.leroux@gmail.com>

bergquist committed:
tech(cloudwatch): refactoring

bergquist committed:
feat(alerting): validate alerts before saving dashboard (#6691)

ref #6576

bergquist committed:
docs(http_api): convert to new docs format

Commit:
Sort series by yaxis by default (#6683)

* Sort series by yxis by default

* Sort series by yxis by default

* Sort series by yxis by default

torkelo committed:
fix(unsaved changes): You now navigate to the intended page after saving in the unsaved changes dialog fixes #6675

bergquist committed:
fix(dashboard): fixes bug when saving dashboards

bergquist committed:
docs(alerting): add info about alerting metrics

torkelo committed:
fix(graph): Fixed tooltip date header visible when x-mode was set to series, fixes #6364

daniellee committed:
feat(changelog): adds TLS Client auth

bergquist committed:
Merge pull request #6692 from grafana/feature/tls-client-auth

Feature/tls client auth

bergquist committed:
docs(changelog): add note about solving sync problem for alerts

bergquist committed:
tech(cloudwatch): store keys in secure json blob

bergquist committed:
feat(cloudwatch): make it possible to reset keys

closes #6697

bergquist committed:
docs(graph): moves thresholds to display tab

closes #6703

bergquist committed:
docs(graph): remove unused headers

bergquist committed:
fix(alerting): fixes broken no_value evaluator

closes #6701

bergquist committed:
docs(changelog): add note about closing #6701

bergquist committed:
docs(http): moves overview content to /http_api/

ref #6719

bergquist committed:
docs(datasource): add root datasource page

bergquist committed:
tech(docs): use relref for links

dpavlos committed:
[6679] Check if org exists before delete it

torkelo committed:
fix(css): minor style fix

bergquist committed:
tech(docker): adds docker fig for elastic 5

bergquist committed:
fix(config): error message for broken config file

closes #6731

bergquist committed:
style(docker): remove comments

bergquist committed:
fix(table): renderes empty date column as '-' (#6736)

* fix(table): renderes empty date column as '-'

closes #6728

* docs(changelog): add note about closing 6728

bergquist committed:
docs(readme): bump golang version

ryantxu committed:
Merge remote-tracking branch 'grafana/master'

ryantxu committed:
use ms when query less then 5 mins

dirk-leroux committed:
Fixed spelling error

dpavlos committed:
[5443] Check if title is empty before save a new dashboard

torkelo committed:
updated version in grafana-server main.go

bergquist committed:
feat(metrics): change active alerts to guage

bergquist committed:
tech: upgrade package json to v4.0.0

bergquist committed:
tech(build): change windows build branch

bergquist committed:
tech(build): update branch to package

torkelo committed:
fix(row): fixed dropping new panels onto panels, should split target panel

torkelo committed:
fix(row): fixed dropping new panels onto panels, should split target panel

bergquist committed:
tech(build): change windows build branch

bergquist committed:
tech(build): set master to 4.1.0 pre

bergquist committed:
tech(build): update publish script for 4.0.x release

bergquist committed:
docs(v4): update docs for v4

bergquist committed:
Merge pull request #6722 from dpavlos/fix_6679

[6679] Check if org exists before delete it

bergquist committed:
docs(changelog): add note about fixing #6679

bergquist committed:
style(sqlstore): remove comment

bergquist committed:
Merge pull request #6741 from dpavlos/fix_5443

[5443] Check if title is empty before save a new dashboard

bergquist committed:
docs(changelog): add note about closing #5443

bergquist committed:
Merge pull request #6670 from tomkozlowski/feature/postgres-certs

generalized database connection cert support and added to postgres

bergquist committed:
docs(changelog): add note about closing #6655

bergquist committed:
docs: remove link to CLA since we use CLA assistance

bergquist committed:
Merge branch 'victorops' of https://github.com/ichekrygin/grafana into ichekrygin-victorops

bergquist committed:
chore(victorops): remove unused code

bergquist committed:
Merge branch 'ichekrygin-victorops'

bergquist committed:
docs: add note about closing #6411

bergquist committed:
Merge pull request #6740 from dirk-leroux/new_aggregation_on_singlestat_panel

New aggregation on singlestat panel

bergquist committed:
docs: add note about closing #6740

bergquist committed:
fix(singlestat): align width for gauge settings

bergquist committed:
fix(singlestat): align width for gauge settings

bergquist committed:
Merge pull request #6698 from grafana/cloudwatch_configurable_keys

Cloudwatch configurable keys from config page.

bergquist committed:
docs: add note about closing cloudwatch

bergquist committed:
Merge pull request #5677 from bmundt/remove_columns

Added Hidden Column Style for Table Panel

bergquist committed:
docs: adds note about closing #5677

fg2it committed:
swap member declaration in StandardGauge to avoid problem with atomic on arm and x86-32

fg2it committed:
swap member declaration in StandardGauge to avoid problem with atomic on arm and x86-32

solugebefola committed:
typo?

bergquist committed:
Merge pull request #6783 from solugebefola/patch-1

typo?

bergquist committed:
Merge pull request #6768 from fg2it/fix6749

fix 6749 by swapping member in StandardGauge to avoid problem with atomic

bergquist committed:
docs(changelog): add note about closing #6749

hvnsweeting committed:
Use space to separate multiple modes

hvnsweeting committed:
Update sample.ini

Use space to separate multiple modes

stuartnelson3 committed:
Add transport cache for reverse proxy (#6776)

* Add transport cache for reverse proxy
* Use updated at in cache

stuartnelson3 committed:
Add transport cache for reverse proxy (#6776)

* Add transport cache for reverse proxy
* Use updated at in cache

bergquist committed:
docs(changelog): note about closing #6759

bergquist committed:
Merge pull request #6796 from hvnsweeting/patch-2

Update sample.ini

bergquist committed:
fix(webdav): adds missing auth headers

closes #6779

bergquist committed:
update to version 4.0.1

bergquist committed:
docs: update links to 4.0.1

torkelo committed:
Update CHANGELOG.md

torkelo committed:
Update CHANGELOG.md

bergquist committed:
update latest.json to 4.0.1

bergquist committed:
fix(influxdb): align tsdb.influxdb with dataproxy

ref #6784

bergquist committed:
change to golang 1.7.4

bergquist committed:
feat(webhook): adds alert rule message to webhook

close #6807

bergquist committed:
feat(webhook): adds alert rule message to webhook

close #6807

bergquist committed:
tech(): replace spaces to handle input more graceful

bergquist committed:
feat(alerting): calculate avg of valid points

bergquist committed:
Merge pull request #6795 from hvnsweeting/patch-1

Use space to separate multiple modes

bergquist committed:
docs(readme): remove link to old CLA page

bergquist committed:
changelog: add note about closing #6807

bergquist committed:
changelog: add bugfix header

bergquist committed:
webhook: adds json content-type

closes #6822

bergquist committed:
webhook: adds json content-type

closes #6822

bergquist committed:
changelog: adds note about closing #6822

Commit:
Use correct domain in render.js (#6813)

Phantomjs is always rendering login page because domain is not considered. According to http://phantomjs.org/api/webpage/method/add-cookie.html a correct domain must be passed. Probably this change is now necessary due to https://github.com/grafana/grafana/issues/6660

torkelo committed:
fix(rendering): minor fix for server side rendering when using non default http addr setting and non default domain setting, #6813

torkelo committed:
Merge branch 'v4.0.x'

torkelo committed:
Merge branch 'master' of github.com:grafana/grafana

torkelo committed:
fix(influxdb): Fix for Ad-Hoc Filters variable & changing dashboards, fixes #6821

torkelo committed:
Merge branch 'v4.0.x'

torkelo committed:
fix(rendering): fixed server side rendering issue when sertting enforce_domain to true, fixes #6769

torkelo committed:
Merge branch 'v4.0.x'

torkelo committed:
updated changelog

torkelo committed:
feat(playlist): added support for kiosk mode & minor polish/fix, closes #6727

torkelo committed:
Merge branch 'v4.0.x'

torkelo committed:
updated changelog

jsoriano committed:
Add 'Last 2 days' option to quick ranges (#6758)

torkelo committed:
change(Timepicker): removed last 90 days to make room for Last 2 days, #6758

torkelo committed:
cleanup(): removed unused (old) migrate stuff

torkelo committed:
fix(logging): added extra logging when default config file is not found (ie homepath is not correct), fixes #6829

torkelo committed:
fix(timerange): fixed time range zoom out shortcut, fixes #6837

torkelo committed:
Merge branch 'v4.0.x'

torkelo committed:
updated changelog

amuraru committed:
Dashboard PNG export (#6644)

torkelo committed:
feat(alerting): added confirm text when removing panel with alert rule

mtanda committed:
(cloudwatch) percentile support (#6634)

* support extended statistics

* handle different response format for extended statistics

mtanda committed:
(cloudwatch) don't expand variable when panel or row repeat is used (#6618)

* (cloudwatch) don't expand variable when panel or row repeat is used

* fix test

Commit:
Only take bar series to calculate barWidth (#6756)

* Only take bar series to calculate barWidth

solves #6751

* Typo

* Typo

* Typo

* Typo

* Refactor

* Simplify

jifwin committed:
Disable web-security for phantomjs rendering (#6469)

torkelo committed:
Merge branch 'master' of https://github.com/ryantxu/grafana into ryantxu-master

torkelo committed:
change(influxdb): always specify absolute time range in milliseconds, refactoring #6571

torkelo committed:
Merge branch 'master' of github.com:grafana/grafana

bergquist committed:
docs: adds note about closing #6779

bergquist committed:
metrics: 64bit fields should come first in structs

https://golang.org/pkg/sync/atomic/#pkg-note-BUG

torkelo committed:
Merge branch 'shared_tooltip' of https://github.com/connection-reset/grafana into connection-reset-shared_tooltip

bergquist committed:
docs: explains how to use comment chars in passwords

torkelo committed:
feat(graph): refactoring shared tooltip PR #6274

torkelo committed:
changelog(): updated changelog with info on shared tooltip feature merged via PR #6274, closes #1578

torkelo committed:
Merge branch 'master' of github.com:grafana/grafana

utkarshcmu committed:
Redo helptext for Grafana v4

utkarshcmu committed:
Redo helptext for Grafana v4

utkarshcmu committed:
Merge branch 'helptext' of https://github.com/utkarshcmu/grafana into helptext

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=4224) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=4224) before we can accept your contribution.<br/>**31** out of **32** committers have signed the CLA.<br/><br/>:white_check_mark: utkarshcmu<br/>:white_check_mark: benrubson<br/>:white_check_mark: mtanda<br/>:white_check_mark: torkelo<br/>:white_check_mark: bergquist<br/>:white_check_mark: stuartnelson3<br/>:white_check_mark: jifwin<br/>:white_check_mark: fg2it<br/>:white_check_mark: amuraru<br/>:white_check_mark: silvestre-abruzzo<br/>:white_check_mark: jsoriano<br/>:white_check_mark: hvnsweeting<br/>:white_check_mark: connection-reset<br/>:white_check_mark: daniellee<br/>:white_check_mark: joelanford<br/>:white_check_mark: dpavlos<br/>:white_check_mark: dirk-leroux<br/>:white_check_mark: ryantxu<br/>:white_check_mark: dirkleroux<br/>:white_check_mark: DanCech<br/>:white_check_mark: peregrinogris<br/>:white_check_mark: mattttt<br/>:white_check_mark: tomkozlowski<br/>:white_check_mark: bfhobbes<br/>:white_check_mark: ichekrygin<br/>:white_check_mark: huydx<br/>:white_check_mark: rogerswingle<br/>:white_check_mark: nicr9<br/>:white_check_mark: rbak1<br/>:white_check_mark: TerraTech<br/>:white_check_mark: msound<br/>:x: solugebefola<hr/>**huydx** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

utkarshcmu committed:
Merge branch 'helptext' of https://github.com/utkarshcmu/grafana into helptext
